### PR TITLE
fix: GLV subscalar range checks

### LINKED ISSUE #1767: fix: bound the subscalar widths in GLV methods
There is an old soundness bug still in emulated GLV/fakeGLV/GLV+fakeGLV -- when the hint returns the subscalars, we check that they add up to the initial scalar. But the issue is that we only use subset of the bits when doing scalar mul (corresponding to the expected non-zero bits).

Assuming that the hint is valid and does lattice reconstruction, then the upper bits should always be zero, but we cannot assume an honest prover.

For example:
```
r = P-256 scalar order
n = 128
H = 2^128
s = r - H - 1

malicious decomposition:
  s1 = H + 1
  s2 = 1
  sign = 0

malicious scalar-mul output:
  R = [r - 1]P = -P
The full scalar relation passes:

s1 + s*s2 = (H+1) + (r-H-1) = r = 0 mod r
But the accumulator only uses the low 128 bits:

low128(s1) = 1
low128(s2) = 1
So it checks:

[1]P + [1]R = P - P = O
while the honest output should be [r-H-1]P, not -P
```

We should address it consistently over all curves and G1/G2. I think the best approach for solving it is actually in non-native package. We currently anyway range check the outputs coming from the hint to be in range (of the non-native modulus), but could optionally make the bound smaller. This actually should reduce the constraints.

_Originally posted by @ivokub in https://github.com/Consensys/gnark/pull/1763#pullrequestreview-4267621967_

### PR DESCRIPTION
**Pull request by ivokub**
# Description

Fixes #1767 

Also implements in generic hint calling in field emulation to define the range check amounts. I think this is also independently relevant elsewhere where we expect smaller outputs from the hint.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How has this been tested?

Existing tests for scalarmul pass

Added tests for the reduced rangecheck edge cases in field emulation.

# Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I did not modify files generated from templates
- [x] `golangci-lint` does not output errors locally
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

---

> [!NOTE]
> **Medium Risk**
> Changes in-circuit soundness assumptions for GLV/fake-GLV hints—incorrect bit bounds could admit malicious decompositions—though bounds match documented lattice limits and are covered by new tests.
> 
> **Overview**
> Adds **`WithHintOutputRangeCheckBits`** to emulated generic hints so hint outputs can be range-checked and packed at a **custom bit width** (via `packLimbsWithWidth` / refactored `enforceWidth`), instead of always using full modulus-width checks.
> 
> **GLV / fake-GLV scalar multiplication** (emulated `sw_emulated`, native G2 on BLS12-381/BN254/BW6-761, Grumpkin) now passes **LLL/Hermite-sized bounds** on decomposed sub-scalars (`u1,u2,v1,v2`, `s1,s2`, etc.) when calling `NewHintGeneric`, aligning constraints with proven decomposition sizes and **reducing scalar-mul cost** (reflected in `latest_stats.csv`).
> 
> Removes **unused** `decomposeScalarG1` / `decomposeScalarG2` hint registrations where decomposition moved to shared emulated paths; Grumpkin’s `callDecomposeScalar` drops the `simple` flag and uses the generic hint API with **127-bit** output checks.
> 
> New **`field_hint_test`** coverage exercises custom range-check indexing, multi-limb widths, and validation errors.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 852a650137c22a9821fec8b358b648b822ecf8f9. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: std/math/emulated/field_hint.go [hunk 1]
```diff
@@ -334,13 +355,49 @@ func unwrapGenericHintOutputs[T1, T2 FieldParams](field *big.Int, fp1 *Field[T1]
...
+		// we only return the element with custom number of limbs so the caller
+		// can never access the potentially non-zero limbs beyond nbCustomLimbs,
+		// and thus there is no soundness issue.
+		outputs[idx] = fp.packLimbsWithWidth(limbs[:nbCustomLimbs], bits)
+	} else {
+		// bits <= 0: pack with default width but without range check.
+		// however, return it as non-internal so that when it is used as input to another operation,
+		// it will be properly range checked.
+		outputs[idx] = &Element[T]{Limbs: limbs, overflow: 0, internal: false}
+		// outputs[idx] = fp.newInternalElement(limbs, 0)
```
Comment (yelhousni) [inline std/math/emulated/field_hint.go]: dead comment?

File: std/algebra/emulated/sw_emulated/point.go [hunk 2]
```diff
@@ -1372,7 +1380,9 @@ func (c *Curve[B, S]) scalarMulFakeGLV(Q *AffinePoint[B], s *emulated.Element[S]
 
 	// First we find the sub-salars s1, s2 s.t. s1 + s2*s = 0 mod r and s1, s2 < sqrt(r).
```
Comment (yelhousni) [inline std/algebra/emulated/sw_emulated/point.go]: comment should be updated to |s1|,|s2| < γ₂·√r ≈ 1.15·√r

**Review by yelhousni — DISMISSED** [review]
LGTM. The bound tightening is mathematically sound — (BitLen+3)/4 + 2 covers γ₄·r^(1/4) and (BitLen+1)/2 covers γ₂·√r with comfortable slack. The WithHintOutputRangeCheckBits API is a clean generalization and the enforceWidth refactor preserves existing semantics. Test coverage (multi-limb, zero-bits, bad indices) is good.

File: std/math/emulated/field_hint.go (hunk 1 above)
Comment (ivokub) [reply std/math/emulated/field_hint.go]: Yup, removed

File: std/algebra/emulated/sw_emulated/point.go (hunk 2 above)
Comment (ivokub) [reply std/algebra/emulated/sw_emulated/point.go]: Updated comment.
